crates/mem-pool/src/pool.rs

```diff
@@ -1062,6 +1063,7 @@ impl MemPool {
                     unused_withdrawals.push(withdrawal_hash);
                 }
             }
+            db.insert_mem_pool_withdrawal(&withdrawal.hash().into(), withdrawal)?;
```

Comment:
We should only re-insert withdrawal in `Ok` branch, in error branch the withdrawal is dropped.

Reply:
Updated.